Tinystories

huevosabio opened the pull request.

huevosabio committed:
tiny stories config

huevosabio committed:
Merge branch 'dotconfig' into tinystories

huevosabio committed:
added a tiny model to test locally the whole pipeline, changed the runner to be based on the config rather than hard coded datasets, fixed the lora finetuner to follow the quant options in the config, added minor readme instructions

huevosabio commented:
adds some minor changes:

1.  a new config for a tiny model just to test out functionality locally
2. git ignore to ignore catching all the outputs from the adapter folder
3. instructions on getting the weights from hf
4. changes to the lora finetuner to respect quantization choices
5. changes to lora finetuner to make the output path if it doesn't exist
6. a handful of pip requirements that I needed for it to run locally

huevosabio commented:
also, this already includes the changes from #4

huevosabio committed:
now going to test tiny_qlora on a single gpu

sumo43 commented:
I think we are just using timdettmers guanaco for testing but I'll keep this open in case that changes

huevosabio committed:
merged

huevosabio committed:
checked that it works locally, moving on to gpus

huevosabio committed:
config changed on tiny qlora

huevosabio committed:
removed a print

huevosabio committed:
increasing batches, printing state

huevosabio committed:
bfloat stuf

huevosabio committed:
removed a line that really does nothing

prateeky2806 committed:
Integrated dataset.
added special token code from qlora.
setting different precisons for different layers. lm_head, norm, etc.
removed some hardcoded parameters.
updated readme

prateeky2806 committed:
Added dataset specific yaml files
Updates requirements.txt to spcify datasets versions.
added camel and sharegpt datasets.

huevosabio committed:
merging latest setup

huevosabio committed:
merging

huevosabio committed:
merged pr8
